Plugins/Rhetos.RestGenerator/InitialCodeGenerator.cs

```diff
@@ -182,11 +183,27 @@ public void InitializeApplicationInstance(System.Web.HttpApplication context)
 
     public static class RestServiceMetadata
     {{
+        {RestServiceMetadataMembersTag}
+
         // This is separated from generic rest service class, because a static field in a generic type is not shared among instances of different close constructed types.
-        public static readonly IReadOnlyDictionary<string, Tuple<string, Type>[]> FilterTypesByDataStructure = new Dictionary<string, Tuple<string, Type>[]>
+        private static ConcurrentDictionary<string, Tuple<string, Type>[]> FilterTypesByDataStructure = new ConcurrentDictionary<string, Tuple<string, Type>[]>();
+
+        public static Tuple<string,Type>[] GetFilterTypesByDataStructure(string dataStructureName)
         {{
-            {FilterTypesByDataStructureTag}
-        }};
+            if(FilterTypesByDataStructure.TryGetValue(dataStructureName, out Tuple<string,Type>[] value))

```

Comment:
Please use `GetOrAdd` instead of `TryGetValue` and `AddOrUpdate`. Then we can remove `ErrorOnUpdate`.